### PR TITLE
use anyio for tests

### PR DESCRIPTION
**Pull request by altendky**
### Purpose:

If we do this, then we can use anyio in code being tested...

Especially useful in conjunction with https://github.com/Chia-Network/chia-blockchain/pull/16670.

### Testing Notes:

```bash
find tests/ -iname 'test_*.py' -exec sed -i 's/pytest_asyncio\.fixture/pytest.fixture/' {} \;
find tests/ -iname 'conftest.py' -exec sed -i 's/pytest_asyncio\.fixture/pytest.fixture/' {} \;
find tests/ -iname 'test_*.py' -exec sed -i 's/pytest\.mark\.asyncio/pytest.mark.anyio/' {} \;
find tests/ -iname 'conftest.py' -exec sed -i 's/pytest\.mark\.asyncio/pytest.mark.anyio/' {} \;
```

### CODE REVIEW

File: tests/wallet/dao_wallet/test_dao_clvm.py [hunk 1]
```diff
@@ -1226,7 +1226,7 @@ async def do_spend(
     return result
 
 
-@pytest.mark.asyncio()
+@pytest.mark.anyio()
```
Comment (AmineKhaldi) [inline tests/wallet/dao_wallet/test_dao_clvm.py:1229]: Maybe just `@pytest.mark.anyio` ?

File: tests/wallet/clawback/test_clawback_lifecycle.py [hunk 2]
```diff
@@ -84,7 +84,7 @@ def sign_coin_spend(self, coin_spend: CoinSpend, index: int) -> G2Element:
             signatures.append(signature)
         return AugSchemeMPL.aggregate(signatures)
 
-    @pytest.mark.asyncio()
+    @pytest.mark.anyio()
```
Comment (AmineKhaldi) [inline tests/wallet/clawback/test_clawback_lifecycle.py:87]: Maybe just `@pytest.mark.anyio` ?

File: tests/wallet/nft_wallet/test_nft_lifecycle.py [hunk 3]
```diff
@@ -26,7 +26,7 @@
 ACS_PH = ACS.get_tree_hash()
 
 
-@pytest.mark.asyncio()
+@pytest.mark.anyio()
```
Comment (AmineKhaldi) [inline tests/wallet/nft_wallet/test_nft_lifecycle.py:29]: Maybe just `@pytest.mark.anyio` ?

File: tests/wallet/nft_wallet/test_nft_lifecycle.py [hunk 4]
```diff
@@ -136,7 +136,7 @@ async def test_state_layer(cost_logger: CostLogger, metadata_updater: str) -> No
             state_layer_puzzle = create_nft_layer_puzzle_with_curry_params(metadata, METADATA_UPDATER_PUZZLE_HASH, ACS)
 
 
-@pytest.mark.asyncio()
+@pytest.mark.anyio()
```
Comment (AmineKhaldi) [inline tests/wallet/nft_wallet/test_nft_lifecycle.py:139]: Maybe just `@pytest.mark.anyio` ?

File: tests/wallet/nft_wallet/test_nft_lifecycle.py [hunk 5]
```diff
@@ -244,7 +244,7 @@ async def test_ownership_layer(cost_logger: CostLogger) -> None:
         ).get_tree_hash()
 
 
-@pytest.mark.asyncio()
+@pytest.mark.anyio()
```
Comment (AmineKhaldi) [inline tests/wallet/nft_wallet/test_nft_lifecycle.py:247]: Maybe just `@pytest.mark.anyio` ?

**Review by AmineKhaldi — DISMISSED** [review]

**Review by AmineKhaldi — DISMISSED** [review]

**Review by emlowe — DISMISSED** [review]

**Review by AmineKhaldi — APPROVED** [review]

**Review by emlowe — APPROVED** [review]

**Pull request merged**